src/node_dotenv.cc

```diff
@@ -168,3 +175,797 @@ void Dotenv::ParseLine(const std::string_view line) {
 }
 
 }  // namespace node
+
+
+namespace cppnv {
+VariablePosition::VariablePosition(const int variable_start,
+                                   const int start_brace,
+                                   const int dollar_sign)
+  : variable_start(
+        variable_start),
+    start_brace(start_brace),
+    dollar_sign(dollar_sign),
+    end_brace(0),
+    variable_end(0) {
+  variable_str = new std::string();
+}
+cppnv::EnvStream::EnvStream(std::string* data) {
+  this->data_ = data;
+  this->length_ = this->data_->length();
+  this->is_good_ = this->index_ < this->length_;
+}
+
+char cppnv::EnvStream::get() {
+  if (this->index_ >= this->length_) {
+    return -1;
+  }
+  const auto ret = this->data_->at(this->index_);
+  this->index_++;
+  this->is_good_ = this->index_ < this->length_;
+  return ret;
+}
+
+bool cppnv::EnvStream::good() const {
```

Comment:
The naming convention makes it really hard to understand what this function does.